src/rgw/rgw_admin.cc

```diff
@@ -3322,7 +3322,7 @@ int main(int argc, const char **argv)
         return EINVAL;
       }
     } else if (ceph_argparse_witharg(args, i, &val, "--max-size", (char*)NULL)) {
-      max_size = strict_iec_cast<long long>(val.c_str(), &err);
+      max_size = strict_iec_cast<long long>(val, &err);
```

Comment:
Should we squash this commit with the previous one? Is the buildability preserved between them?

Reply:
indeed. folded.